### ISSUE TITLE
Wrong behavior from the Flyout component in Wazuh Dashboard 2.x

### ISSUE THREAD

**Issue by AlexRuiz7** [wazuh/wazuh-kibana-app#4597]
## Description
A new problem has been found in every version of OpenSearch 2.x, more concretely related to the OpenSearch Dashboards. 

We use the [flyout](https://elastic.github.io/eui/#/layout/flyout) component from EUI, which closes automatically when the user clicks outside it. However, in 2.x, these components will also close automatically when clicking inside them. This is not the correct behavior, and it's probably related to OSD updating the EUI library to a newer version, breaking our customized components behavior.

<details>
<summary>Video</summary>

https://user-images.githubusercontent.com/63758389/191055783-8a658ce0-0bc4-466f-a050-f672f7a45360.mp4

</details>

We'll need to hunt and fix this bug.

### Modules

- [x] Management - Rules
- [x] Mitre Attack - Intelligence
- [x] Mitre Attack - Techniques
- [x] Agents - FIM - Inventory
- [x] All Compliance - Controls Tab - Requirements Table

### Security

- [x] Policies - Create and Edit
- [x] Roles Mapping - Create and Edit
- [x] Roles - Create and Edit
- [x] User - Create and Edit

### Management

- [x] Decoders - Detail Flyout
- [x] Ruleset - Detail Flyout and Rule Info

**Related release notes**

> OpenSearch Dashboards uses [OUI](https://github.com/opensearch-project/oui) and its alias onto EUI (https://github.com/opensearch-project/OpenSearch-Dashboards/pull/2080)
https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/release-notes/opensearch-dashboards.release-notes-2.2.0.md

> Bumps @elastic/eui to v34.6.0 and @elastic/charts to v31.1.0
https://github.com/opensearch-project/OpenSearch-Dashboards/blob/main/release-notes/opensearch-dashboards.release-notes-2.0.0.md

_Originally posted by @yenienserrano in https://github.com/wazuh/wazuh-kibana-app/issues/4160#issuecomment-1251188695_

**Comment by Machi3mfl** [comment]
### Issue research

![image](https://user-images.githubusercontent.com/6089438/192835552-9dbb08c2-d0f6-4530-9ef9-6d94d8fb23b6.png)

![Selection_044](https://user-images.githubusercontent.com/6089438/192834290-1ebeeba0-d0bb-498a-8a3a-d5e75cde8059.png)

![image](https://user-images.githubusercontent.com/6089438/192836092-ad3467dd-fa9e-4e06-abc0-2e5868bee79f.png)

**Comment by AlexRuiz7** [comment]
The `WzFlyout` component wraps the `EuiFlyout` component with an `EuiOutsideClickDetector` component (to detect clicks outside the flyout) and an `EuiOverlayMask` component, to dim the area outside the flyout.

These 2 functionalities are now provided out-of-the-box by the [`EuiFlyout` component](https://elastic.github.io/eui/#/layout/flyout), so we'll test whether using this component alone solves the issue, in such case we'll replace the `WzFlyout` component, and remove it from the code.

https://github.com/wazuh/wazuh-kibana-app/blob/370b09a2bb3c33b2f0a6d4a5080c673d2980f163/public/components/common/flyouts/wz-flyout.tsx#L17-L33

**Comment by Machi3mfl** [comment]
## Sections involved

### Modules
- Management - Rules
- Mitre Attack - Intelligence
- Mitre Attack - Techniques
- Agents - FIM - Inventory
- All Compliance - Controls Tab - Requirements Table

### Security
- Policies - Create and Edit
- Roles Mapping - Create and Edit
- Roles - Create and Edit
- User - Create and Edit

### Management
- Decoders - Detail Flyout
- Ruleset - Detail Flyout and Rule Info

**Comment by AlexRuiz7** [comment]
During the review of the #4638, @Machi3mfl and I discovered that there are important layout differences between our Kibana app and the Wazuh Dashboard app in some sections that are affected by this issue.

For example, the ruleset table uses a page component in Kibana, while our WzD app uses a flyout component.

![image](https://user-images.githubusercontent.com/15186973/196499223-2c529e48-ed64-41e1-a9f6-c6dc24975fda.png)

![image](https://user-images.githubusercontent.com/15186973/196499242-ac61c258-510c-4312-b59c-70023ad42f59.png)

We concluded that the work from #4174 by @asteriscos was not properly ported to the dashboard app. We need to review this port. Until we arrange this situation, this issue be **blocked**.
